cmd/operator/service.go

```diff
@@ -69,23 +77,29 @@ func getServicesByRegionAndLabel(ctx context.Context, project, region, labelSele
 		return nil, errors.Wrapf(err, "failed to get services with label %q in region %q", labelSelector, region)
 	}
 
+	lg.Debugf("found %d service(s)", len(svcs))
 	return svcs, nil
 }
 
 // determineRegions gets the regions the label selector should be searched at.
 //
 // If the target configuration does not specify any regions, the entire list of
 // regions is retrieved from API.
-func determineRegions(ctx context.Context, target *config.Target) ([]string, error) {
+func determineRegions(ctx context.Context, logger *logrus.Logger, target *config.Target) ([]string, error) {
 	regions := target.Regions
 	if len(regions) != 0 {
+		logger.Debug("using predefined list of regions, skip querying from API")
 		return regions, nil
 	}
 
+	logger.Debug("retrieving all regions from the API")
+
 	regions, err := runapi.Regions(ctx, target.Project)
 	if err != nil {
-		return nil, errors.Wrap(err, "Cannot get list of regions from Cloud Run API")
+		return nil, errors.Wrap(err, "cannot get list of regions from Cloud Run API")
 	}
+
+	logger.Debugf("finished retrieving %d regions from the API", len(regions))
```

Comment:
don't have to fix here but in the future you can do this:

logger.Withfield("n",len(...)).Debugf(...)

technically any %x can be a field :)